src/triage/experiments/base.py

```diff
@@ -137,6 +142,26 @@ def __init__(
         self.features_schema_name = "features"
         self.materialize_subquery_fromobjs = materialize_subquery_fromobjs
         self.features_ignore_cohort = features_ignore_cohort
+
+
+        ## Defaults to sane values
+        self.config['temporal_config'] = self._fill_timechop_config_missing()
+        ## Defaults to all the entities found in the features_aggregation's from_obj
+        self.config['cohort_config'] = self._fill_cohort_config_missing()
+        ## Defaults to all the feature_aggregation's prefixes
+        self.config['feature_group_definition'] = self._fill_feature_group_definition()
+
+        # if using a model grid preset, fill in the actual grid
+        if self.config.get('model_grid_preset'):
```

Comment:
For consistency, this check seems like it would make sense in experiments/validate.py instead.

Reply:
That's fair, though we're doing the filling before running validation (since validation will fail on the missing values otherwise). Maybe that suggests we should modify the validation scripts to accept these missing values, but I think we still want to validate the full config post-filling regardless, so there's a little bit of a chicken-and-egg problem...

Reply:
You're right, it does require some thinking about how the validation flow mixes with the filling flow.  My viewpoint is that the 'filling' can just replace the preset grid with the expanded one, and *remove the preset key*. That way, the validation actually wouldn't change it all: it doesn't know what a preset is, just that something along the way put in a real grid.

Expanding this line of thinking further would actually just remove this filling logic from the Experiment entirely. It's just a function that a user may call to transform a truncated version of experiment config into a full one. Based on the goal of a simpler user interface, this may not be what you want, of course.

Maybe this means that the 'config transformer' needs its own round of validation, which is more in line with what you put in there; though even if this is true I still maintain it looks kind of awkward just sitting there in the Experiment's constructor. This is another thing that separating out this logic in to its own module would help with; if this check was sitting at the top of a function that just transformed the experiment config, it would have looked less awkward to me.

Reply:
That makes sense and lines up with the way we were thinking about it (that the validator should act on the final config after any default filling). We moved the logic out into `defaults.py` here, so hopefully that looks better now.